### PR TITLE
Get Options by IDs from dataset API

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Upgraded `dp-api-clients-go` dependency to v1.32.0 in order to match filter API response with `GetDimensions` pagination

Improved efficiency by batch processing where needed, and requesting only the necessary information from APIs where possible:
- getIDNameLookupFromDatasetAPI now gets only the required IDs from dataset API (in batches of `MaxDatasetOptions`) instead of getting all options in batches
- generalised dataset API batch processing (`BatchProcessDimensionOptionsFromDatasetAPI`):
  - used to accumulate results in GetDimensionOptionsFromDatasetAPI
  - used in dimensions.go addAll to do the dataset.Get+filter.Patch(add) options in batches if needed
- time.go, age.go and dimensions.go now get the count of allValues before deciding if we need to continue, and get all values (in batches) as late as possible, only if we do need them.
- Refactored NewFilter to obtain the new `MaxDatasetOptions` value from config

### How to review

- Make sure code changes make sense
- Make sure unit tests pass

### Who can review

Anyone

### CODE REVIEW

File: handlers/common.go [hunk 1]
```diff
@@ -91,63 +101,112 @@ func (f *Filter) getIDNameLookupFromDatasetAPI(ctx context.Context, userAccessTo
...
+	// function to aggregate items
+	processBatch := func(batch filter.DimensionOptions) error {
+		if batch.Offset == 0 {
+			opts.TotalCount = batch.TotalCount
+		}
+		opts.Items = append(opts.Items, batch.Items...)
+		return nil
+	}
+
+	// call filter API GetOptions in bathes and aggregate the responses
```
Comment (CarlHembrough) [inline handlers/common.go:113]:
```suggestion
	// call filter API GetOptions in batches and aggregate the responses
```

File: handlers/common.go [hunk 2]
```diff
@@ -91,63 +101,112 @@ func (f *Filter) getIDNameLookupFromDatasetAPI(ctx context.Context, userAccessTo
...
+	// function to aggregate items
+	processBatch := func(batch dataset.Options) error {
+		if batch.Offset == 0 {
+			opts.TotalCount = batch.TotalCount
+		}
+		opts.Items = append(opts.Items, batch.Items...)
+		return nil
+	}
+
+	// call dataset API GetOptions in bathes and aggregate the responses
```
Comment (CarlHembrough) [inline handlers/common.go:167]:
```suggestion
	// call dataset API GetOptions in batches and aggregate the responses
```

File: handlers/dimensions.go [hunk 3]
```diff
@@ -247,36 +246,36 @@ func (f *Filter) DimensionSelector() http.HandlerFunc {
...
 
 		// if there are more than maxNumOptionsOnPage, then we need to use the hierarchy model
-		if isHierarchy && opts.TotalCount > maxNumOptionsOnPage {
+		if isHierarchy && opts.TotalCount > MaxNumOptionsOnPage {
 			f.Hierarchy().ServeHTTP(w, req)
 			return
 		}
 
-		selectedValues, err := f.GetDimensionOptionsFromFilterAPI(req.Context(), userAccessToken, collectionID, filterID, name)
+		dims, err := f.DatasetClient.GetVersionDimensions(req.Context(), userAccessToken, "", collectionID, datasetID, edition, version)
```
Comment (CarlHembrough) [inline handlers/dimensions.go:262]: Not part of this PR - but an easy performance win for page load times could be to make these API requests concurrent.

File: handlers/update-version.go [hunk 4]
```diff
@@ -64,27 +65,28 @@ func (f *Filter) UseLatest() http.HandlerFunc {
...
+				var vals []string
+				for _, val := range batch.Items {
+					vals = append(vals, val.Option)
+				}
+				return f.FilterClient.PatchDimensionValues(req.Context(), userAccessToken, "", collectionID, newFilterID, dim.Name, vals, []string{}, f.BatchSize)
 			}
 
-			if err := f.FilterClient.SetDimensionValues(req.Context(), userAccessToken, "", collectionID, newFilterID, dim.Name, vals); err != nil {
-				log.Event(ctx, "failed to add dimension values", log.ERROR, log.Error(err), log.Data{"filter_id": newFilterID, "dimension": dim.Name})
+			// Call filter API GetOptions in bathes and aggregate the responses
```
Comment (CarlHembrough) [inline handlers/update-version.go:87]: 🙈 
```suggestion
			// Call filter API GetOptions in batches and aggregate the responses
```

**Review by CarlHembrough — CHANGES_REQUESTED** [review]
Changes look good - the best I could do was a few typos 😃

File: handlers/dimensions.go (hunk 3 above)
Comment (DavidSubiros) [reply handlers/dimensions.go:262]: good point! I can do it in this PR

**Review by CarlHembrough — APPROVED** [review]
👍
